lib/plugins/package/lib/packageService.js

```diff
@@ -59,44 +61,115 @@ module.exports = {
       });
   },
 
+  // Inspired by https://github.com/rxaviers/async-pool
```

Comment:
I've looked at the code in the original library and the version above is pretty much the same with a little changes to variables, etc. I think we should either add library as a dependency or make it compliant with MIT license of the original library in a similar way as it was done here: https://github.com/serverless/utils/blob/master/download.js